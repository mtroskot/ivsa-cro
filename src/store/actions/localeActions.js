import { localeActionTypes } from 'src/constants/actionTypes';

export function changeLocale(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE,
    payload: { locale }
  };
}
export function changeLocaleSuccess(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE_SUCCESS,
    payload: { locale }
  };
}
