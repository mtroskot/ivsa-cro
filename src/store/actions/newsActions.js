import { newsActionTypes } from 'src/constants/actionTypes';

export function fetchNews(refreshing = false) {
  return {
    type: newsActionTypes.FETCH_NEWS,
    payload: { refreshing }
  };
}
export function fetchNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.FETCH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
export function fetchNewsError(error) {
  return {
    type: newsActionTypes.FETCH_NEWS_ERROR,
    payload: { error }
  };
}

export function publishNews(date, publishedBy, notice) {
  return {
    type: newsActionTypes.PUBLISH_NEWS,
    payload: { date, publishedBy, notice }
  };
}

export function publishNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.PUBLISH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function updateNews(newsId, date, publishedBy, notice) {
  return {
    type: newsActionTypes.UPDATE_NEWS,
    payload: { newsId, date, publishedBy, notice }
  };
}

export function updateNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.UPDATE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function deleteNews(newsId) {
  return {
    type: newsActionTypes.DELETE_NEWS,
    payload: { newsId }
  };
}

export function deleteNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.DELETE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
