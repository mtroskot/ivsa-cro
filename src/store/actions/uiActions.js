import { uiActionTypes } from 'src/constants/actionTypes';
import { ObjectUtils } from 'src/utils';

export const startAction = (name, params) => {
  if (!ObjectUtils.isObject(params)) {
    throw Error(`type of ${typeof params} is not an object`);
  }
  return {
    type: uiActionTypes.START_ACTION,
    payload: {
      action: {
        name,
        params
      }
    }
  };
};

export const stopAction = name => ({
  type: uiActionTypes.STOP_ACTION,
  payload: { name }
});

export const refreshActionStart = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_START,
  payload: { refreshAction }
});

export const refreshActionStop = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_STOP,
  payload: { refreshAction }
});

export const togglePopupMessage = (message, position) => ({
  type: uiActionTypes.TOGGLE_POPUP_MESSAGE,
  payload: {
    message,
    position
  }
});
