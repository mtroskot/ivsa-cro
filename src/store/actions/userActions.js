import { userActionTypes } from 'src/constants/actionTypes';

export function authenticateUser(email, password) {
  return {
    type: userActionTypes.AUTH_USER,
    payload: { email, password }
  };
}

export const authenticateUserSuccess = (displayName, email, idToken, expiryDate) => {
  return {
    type: userActionTypes.AUTH_USER_SUCCESS,
    payload: { displayName, email, idToken, expiryDate }
  };
};

export const authenticateUserError = error => {
  return {
    type: userActionTypes.AUTH_USER_ERROR,
    payload: { error }
  };
};

export const logout = () => {
  return {
    type: userActionTypes.LOG_OUT,
    payload: {}
  };
};
