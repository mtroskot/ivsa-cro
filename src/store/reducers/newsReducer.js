import { newsActionTypes } from 'src/constants/actionTypes';

//initialState.cachedNews will be a 'object array', e.g {a:{},b:{},c:{}}
export const initialState = {
  cachedNews: {},
  lastUpdate: 0,
  error: null
};

const newsReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case newsActionTypes.FETCH_NEWS_SUCCESS:
    case newsActionTypes.PUBLISH_NEWS_SUCCESS:
    case newsActionTypes.DELETE_NEWS_SUCCESS:
    case newsActionTypes.UPDATE_NEWS_SUCCESS:
      return {
        ...state,
        cachedNews: payload.news,
        lastUpdate: payload.lastUpdate,
        error: null
      };
    case newsActionTypes.FETCH_NEWS_ERROR:
      return {
        ...state,
        error: payload.error
      };
    default:
      return state;
  }
};

export default newsReducer;
