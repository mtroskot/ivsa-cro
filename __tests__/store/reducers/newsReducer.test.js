import newsReducer,{initialState} from 'src/store/reducers/newsReducer';
import * as newsActions from 'src/store/actions/newsActions';

describe('INITIAL_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(newsReducer(beforeActionState, action)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns fetchNewsSuccess state', () => {
    const news = { 1: {}, 2: {} };
    const lastUpdate = 1;
    const startAction = newsActions.fetchNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns publishNewsSuccess state', () => {
    const news = { 1: {}, 2: {} };
    const lastUpdate = 1;
    const startAction = newsActions.publishNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('4returns deleteNewsSuccess state', () => {
    const news = { 1: {}, 2: {} };
    const lastUpdate = 1;
    const startAction = newsActions.deleteNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('5returns updateNewsSuccess state', () => {
    const news = { 1: {}, 2: {} };
    const lastUpdate = 1;
    const startAction = newsActions.updateNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('6returns fetchNewsError state', () => {
    const error = 'error';
    const startAction = newsActions.fetchNewsError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});

describe('FETCH_NEWS_SUCCESS_STATE', () => {
  const beforeActionState = {
    ...initialState,
    cachedNews: { 1: {}, 2: {} },
    lastUpdate: 2
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(newsReducer(beforeActionState, action)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns fetchNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.fetchNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns publishNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.publishNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('4returns deleteNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.deleteNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('5returns updateNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.updateNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('6returns fetchNewsError state', () => {
    const error = 'error';
    const startAction = newsActions.fetchNewsError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});

describe('PUBLISH_NEWS_SUCCESS_STATE', () => {
  const beforeActionState = {
    ...initialState,
    cachedNews: { 1: {}, 2: {} },
    lastUpdate: 2
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(newsReducer(beforeActionState, action)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns fetchNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.fetchNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns publishNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.publishNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('4returns deleteNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.deleteNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('5returns updateNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.updateNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('6returns fetchNewsError state', () => {
    const error = 'error';
    const startAction = newsActions.fetchNewsError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});
describe('DELETE_NEWS_SUCCESS_STATE', () => {
  const beforeActionState = {
    ...initialState,
    cachedNews: { 1: {}, 2: {} },
    lastUpdate: 2
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(newsReducer(beforeActionState, action)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns fetchNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.fetchNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns publishNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.publishNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('4returns deleteNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.deleteNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('5returns updateNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.updateNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('6returns fetchNewsError state', () => {
    const error = 'error';
    const startAction = newsActions.fetchNewsError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});

describe('UPDATE_NEWS_SUCCESS_STATE', () => {
  const beforeActionState = {
    ...initialState,
    cachedNews: { 1: {}, 2: {} },
    lastUpdate: 2
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(newsReducer(beforeActionState, action)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns fetchNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.fetchNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns publishNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.publishNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('4returns deleteNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.deleteNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('5returns updateNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.updateNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('6returns fetchNewsError state', () => {
    const error = 'error';
    const startAction = newsActions.fetchNewsError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});

describe('FETCH_NEWS_ERROR_STATE', () => {
  const beforeActionState = {
    ...initialState,
    error: 'error'
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(newsReducer(beforeActionState, action)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns fetchNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.fetchNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate,
      error: null
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns publishNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.publishNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate,
      error: null
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('4returns deleteNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.deleteNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate,
      error: null
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('5returns updateNewsSuccess state', () => {
    const news = { 3: {}, 4: {} };
    const lastUpdate = 3;
    const startAction = newsActions.updateNewsSuccess(news, lastUpdate);
    const expectedState = {
      ...beforeActionState,
      cachedNews: news,
      lastUpdate,
      error: null
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('6returns fetchNewsError state', () => {
    const error = 'error2';
    const startAction = newsActions.fetchNewsError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(newsReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(newsReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});
