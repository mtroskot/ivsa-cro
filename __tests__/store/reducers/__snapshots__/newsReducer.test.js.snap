// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DELETE_NEWS_SUCCESS_STATE 1returns initial state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 2,
}
`;

exports[`DELETE_NEWS_SUCCESS_STATE 2returns fetchNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`DELETE_NEWS_SUCCESS_STATE 3returns publishNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`DELETE_NEWS_SUCCESS_STATE 4returns deleteNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`DELETE_NEWS_SUCCESS_STATE 5returns updateNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`DELETE_NEWS_SUCCESS_STATE 6returns fetchNewsError state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": "error",
  "lastUpdate": 2,
}
`;

exports[`FETCH_NEWS_ERROR_STATE 1returns initial state 1`] = `
Object {
  "cachedNews": Object {},
  "error": "error",
  "lastUpdate": 0,
}
`;

exports[`FETCH_NEWS_ERROR_STATE 2returns fetchNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_ERROR_STATE 3returns publishNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_ERROR_STATE 4returns deleteNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_ERROR_STATE 5returns updateNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_ERROR_STATE 6returns fetchNewsError state 1`] = `
Object {
  "cachedNews": Object {},
  "error": "error2",
  "lastUpdate": 0,
}
`;

exports[`FETCH_NEWS_SUCCESS_STATE 1returns initial state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 2,
}
`;

exports[`FETCH_NEWS_SUCCESS_STATE 2returns fetchNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_SUCCESS_STATE 3returns publishNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_SUCCESS_STATE 4returns deleteNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_SUCCESS_STATE 5returns updateNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`FETCH_NEWS_SUCCESS_STATE 6returns fetchNewsError state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": "error",
  "lastUpdate": 2,
}
`;

exports[`INITIAL_STATE 1returns initial state 1`] = `
Object {
  "cachedNews": Object {},
  "error": null,
  "lastUpdate": 0,
}
`;

exports[`INITIAL_STATE 2returns fetchNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 1,
}
`;

exports[`INITIAL_STATE 3returns publishNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 1,
}
`;

exports[`INITIAL_STATE 4returns deleteNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 1,
}
`;

exports[`INITIAL_STATE 5returns updateNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 1,
}
`;

exports[`INITIAL_STATE 6returns fetchNewsError state 1`] = `
Object {
  "cachedNews": Object {},
  "error": "error",
  "lastUpdate": 0,
}
`;

exports[`PUBLISH_NEWS_SUCCESS_STATE 1returns initial state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 2,
}
`;

exports[`PUBLISH_NEWS_SUCCESS_STATE 2returns fetchNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`PUBLISH_NEWS_SUCCESS_STATE 3returns publishNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`PUBLISH_NEWS_SUCCESS_STATE 4returns deleteNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`PUBLISH_NEWS_SUCCESS_STATE 5returns updateNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`PUBLISH_NEWS_SUCCESS_STATE 6returns fetchNewsError state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": "error",
  "lastUpdate": 2,
}
`;

exports[`UPDATE_NEWS_SUCCESS_STATE 1returns initial state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": null,
  "lastUpdate": 2,
}
`;

exports[`UPDATE_NEWS_SUCCESS_STATE 2returns fetchNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`UPDATE_NEWS_SUCCESS_STATE 3returns publishNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`UPDATE_NEWS_SUCCESS_STATE 4returns deleteNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`UPDATE_NEWS_SUCCESS_STATE 5returns updateNewsSuccess state 1`] = `
Object {
  "cachedNews": Object {
    "3": Object {},
    "4": Object {},
  },
  "error": null,
  "lastUpdate": 3,
}
`;

exports[`UPDATE_NEWS_SUCCESS_STATE 6returns fetchNewsError state 1`] = `
Object {
  "cachedNews": Object {
    "1": Object {},
    "2": Object {},
  },
  "error": "error",
  "lastUpdate": 2,
}
`;
